client/utils/guidance.py

```diff
@@ -218,10 +236,13 @@ def set_tg(self,access_token,guidance_flag):
                 return EMPTY_MISUCOUNT_TGID_PRNTEDMSG
 
             try:
-                data = json.loads(urlopen(TGSERVER + cur_email + TG_SERVER_ENDING,timeout =1).read().decode("utf-8"))
+                data = json.loads(urlopen(TGSERVER + cur_email +"/" + self.assignment + TG_SERVER_ENDING,timeout =1).read().decode("utf-8"))
             except IOError as e:
                 data = {"tg":-1}
-
+                log.error("Failed to communicate to server. Error: %s", str(e))
+            if(data.get("tg") == None):
```

Comment:
Is this block supposed to be at this indentation level? `e` doesn't appear to be defined anywhere.